src/lib/replace-icon.ts

```diff
@@ -28,8 +28,13 @@ export function replaceIconInRow(
 ): void {
   const fileName = itemRow
     .querySelector(provider.selectors.filename)
-    ?.textContent?.split('/')[0]
-    .trim();
+    ?.textContent// get the last folder for the icon
+    ?.split('/')
+    .reverse()[0]
+    .trim()
+    // remove possible sha from submodule
+    // matches 4 or more to future proof in case they decide to increase it.
+    .replace(/\s+@\s+[a-fA-F0-9]{4,}$/, '');
```

Comment:
I'm wondering if the regular expression is needed here. As far as I understood the logic, we simply want to remove the hash code after the filename. So wouldn't is be possible to use a `split` function for this?

```suggestion
    .split('@')[0].trim()
```

The split divides the filename into two parts: the part before the "@" and the part after it. In this case we're using the first part by using `.split('@')[0]` and trim it afterwards. I don't expect that there's something after the hash code. 

What do you think about this? Am I missing something or can the complexity be reduced a bit by removing the regular expression?

Reply:
@ symbol can exist for a folder name, and since the developer can decide what to call their submodule, it is possible for them to use the symbol as a name for the folder. This is just feature proofing. We know there are currently no icons matching the @ symbol, but who knows what the future holds.

Reply:
I was just thinking, and maybe your approach is safer if the filename itself also includes an "@" symbol, right? 😅 Probably we should keep the regex.

Reply:
I think we commented at the same time, if you want you can wait to see other people's opinions. I am for the regex.

Reply:
Yes, it looks like :) I'm also for the regex approach now!